### ISSUE TITLE
Patch not working w/ xcode 9 ?

### ISSUE THREAD

**Issue by b-lester** [juce/xcs-tweaks#2]
I installed the beta on our build server to get ahead of the ios11 release and the performance issues returned. Looks like the devicesClass.js file has changed since you wrote your patch originally. I am hesitant to apply it. Have you tested it with xcode 9?

Thank you for creating the patch it has been very helpful for us.

**Comment by juce** [comment]
i have not tried it with Xcode 9 yet.
Can you upload deviceClass.js somewhere? I can compare it to Xcode 8's version then
Thanks.

**Comment by JakubMazur** [comment]
Yep, confirmed, does not working anymore. The file is in attachment (3 of them, because not sure what orig is). `rej` one is a product of running a script on it.

Here is an output:

patching file deviceClass.js
```
Hunk #1 FAILED at 12.
Hunk #2 FAILED at 142.
Hunk #3 FAILED at 167.
3 out of 3 hunks FAILED -- saving rejects to file deviceClass.js.rej
```

[deviceClass.zip](https://github.com/juce/xcs-tweaks/files/1317095/deviceClass.zip)

**Comment by ArthurAlvarez** [comment]
I'm facing the same problem

**Comment by dimitar-lazarov** [comment]
Same here

**Comment by Sander-Kornev** [comment]
The same. Someone found solution to simulator issue?

**Comment by ArthurAlvarez** [comment]
The only thing that worked for me was executing the following command:
`sudo xcscontrol --reset`

Be ware that it will delete all your existing bots... But now it's working like a charm here!

**Comment by juce** [comment]
Looks like they changed quite a bit of that logic in Xcode 9. I'll need to do some experiments...
Haven't ran into performance issues yet with the new Xcode though, are people having the same problems still? ("node" processes high CPU usage, unresponsiveness of API, etc.)

**Comment by juce** [comment]
an update on this:
so i was able to make variation of the patch for Xcode 9, but unfortunately i don't think it helps with performance issues. The actual true root cause of the problem is the bloated "all-devices" view in CouchDB. So it might be simpler to just reset XcodeServer, if you don't have a lot of bots and can re-create them with not too much grief...  That did help in my case, and i'm not seeing CPU problems. Looks like Apple did fix the issue with CouchDB "all-devices" view. But in order to benefit from the fix, you either have to reset the server (which nukes the CouchDB database), or if you must preserve the existing database with all the bot builds history, then it is probably possible to manually repair the view. I haven't tried though...
